shared/amms.py

```diff
@@ -99,152 +143,76 @@ async def get_data(self) -> None:
-        self.add_pool(
-            "DAI",
-            "STRK",
-            [
-                "0x048ddb56ceb74777d081a9ce684aaa78e98c286e14fc1badb3a9938e710d6866",  # jediswap
-            ],
-            None,
-        )
-        await self.get_balance()
+    """SwapAmm class for managing multiple liquidity pools."""
+
+    def __init__(self) -> None:
+        """
+        Initialize a SwapAmm instance for managing multiple liquidity pools.
+        """
+        self.pools: Dict[str, Pool] = {}
+        self.add_all_pools()
 
-    async def get_balance(self):
+    def add_all_pools(self) -> None:
+        """
+        Add all predefined pools based on configuration in POOL_MAPPING.
+        """
+        for pool_name, pool_data in POOL_MAPPING.items():
+            self.add_pool(
+                base_token=pool_data["base_token"],
+                quote_token=pool_data["quote_token"],
+                pool_addresses=pool_data["addresses"],
+                myswap_id=pool_data.get("myswap_id"),
+            )
+
+    async def get_balance(self) -> None:
+        """
+        Retrieve balances for each pool's tokens asynchronously.
+        """
         for pool in self.pools.values():
             await pool.get_balance()
 
-    def add_pool(self, base_token: str, quote_token, pool_addresses, myswap_id=None):
+    def add_pool(
+        self,
+        base_token: str,
+        quote_token: str,
+        pool_addresses: List[str],
+        myswap_id: Optional[int] = None,
+    ) -> None:
+        """
+        Add a new liquidity pool for a token pair to the SwapAmm instance.
+
+        :param base_token: The symbol for the base token in the pair.
+        :param quote_token: The symbol for the quote token in the pair.
+        :param pool_addresses: A list of addresses associated with the pool.
+        :param myswap_id: Optional identifier for MySwap pools.
+        """
         if myswap_id is None:
             pool = Pool(base_token, quote_token, pool_addresses, myswap_id)
         else:
             pool = MySwapPool(base_token, quote_token, pool_addresses, myswap_id)
         self.pools[pool.id] = pool
 
-    def get_pool(self, base_token, quote_token):
-        pools = self.pools.get(self.tokens_to_id(base_token, quote_token), None)
-        if not pools:
-            raise ValueError(
-                f"Trying to get pools that are not set: {self.tokens_to_id(base_token, quote_token)}"
-            )
-        return pools
+    def get_pool(self, base_token: str, quote_token: str) -> Pool:
+        """
+        Retrieve a pool instance for a specific token pair.
+
+        :param base_token: The symbol of the base token.
+        :param quote_token: The symbol of the quote token.
+        :return: The Pool instance associated with the token pair.
+        :raises ValueError: If the requested pool is not found.
+        """
+        pool_id = self.tokens_to_id(base_token, quote_token)
+        pool = self.pools.get(pool_id, None)
```

Comment:
if you use `get` it will by deafult return `None` so feel free to remove `None` here